### PR TITLE
Include ADS license to the Warehouse object

### PR DESCRIPTION
**Pull request by mrataj-gd**
resolves [#441](https://github.com/gooddata/gooddata-java/issues/441)

### CODE REVIEW

**Comment by martiner** [comment]
WTF "DO NOT MERGE"? don't open the pull request then...

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
     <groupId>com.gooddata</groupId>
     <artifactId>gooddata-java</artifactId>
-    <version>2.4.3-SNAPSHOT</version>
```
Comment (martiner) [inline pom.xml]: remove the change

File: src/main/java/com/gooddata/warehouse/Warehouse.java [hunk 2]
```diff
@@ -75,7 +77,8 @@ public Warehouse(@JsonProperty("title") String title, @JsonProperty("authorizati
                      @JsonProperty("createdBy") String createdBy, @JsonProperty("updatedBy") String updatedBy,
                      @JsonProperty("status") String status, @JsonProperty("environment") String environment,
                      @JsonProperty("connectionUrl") String connectionUrl,
-                     @JsonProperty("links") Map<String, String> links) {
+                     @JsonProperty("links") Map<String, String> links,
+                     @JsonProperty("license") License license) {
```
Comment (martiner) [inline src/main/java/com/gooddata/warehouse/Warehouse.java]: you can't do such incompatible change in a public constructor

Comment (martiner) [inline src/main/java/com/gooddata/warehouse/Warehouse.java]: moreover you probably haven't read https://github.com/gooddata/gooddata-java/blob/master/CONTRIBUTING.md#enums

**Comment by martiner** [comment]
- [ ] test case which proves deserialization works for old as well as the new version of the server rest api

File: pom.xml (hunk 1 above)
Comment (mrataj-gd) [inline pom.xml]: removed

File: src/main/java/com/gooddata/warehouse/Warehouse.java (hunk 2 above)
Comment (mrataj-gd) [inline src/main/java/com/gooddata/warehouse/Warehouse.java]: created a new constructor and changed `license` type to `String`

**Pull request merged**